MapboxDirections/MBVisualInstructionComponent.swift

```diff
@@ -20,43 +23,47 @@ public class VisualInstructionComponent: NSObject, NSSecureCoding {
+     The scale factor of the image.
+     
+     If you multiply the logical size of the image (stored in the `size` property) by the value in this property, you get the dimensions of the image in pixels.
+     
+     The default value of this property matches the natural scale factor associated with the main screen. However, only images with a scale factor of 1.0, 2.0, or 3.0 are ever returned by the API, so a scale factor of 1.0 of less results in a 1× (standard-resolution) image, while a scale factor greater than 1.0 results in a 2× (high-resolution or Retina) image.
      */
-    #if os(iOS) || os(tvOS)
-        /**
-        A dictionary containing the URLs to image representations of this component.
+    open var scale: CGFloat = NSScreen.main()?.backingScaleFactor ?? 1
+    #elseif os(watchOS)
+    /**
+     The scale factor of the image.
+     
+     If you multiply the logical size of the image (stored in the `size` property) by the value in this property, you get the dimensions of the image in pixels.
      
-        The keys of this dictionary are trait collections specifying different display scales. The values of this dictionary are URLs to remote images that you should prefer over the `text` property once they are available. Use the URL that is best suited to the system’s current traits.
-        */
-        public var imageURLs: [UITraitCollection: URL] = [:]
+     The default value of this property matches the natural scale factor associated with the screen. Images with a scale factor of 1.0, 2.0, or 3.0 are ever returned by the API, so a scale factor of 1.0 of less results in a 1× (standard-resolution) image, while a scale factor greater than 1.0 results in a 2× (high-resolution or Retina) image.
+     */
+    open var scale: CGFloat = WKInterfaceDevice.current().screenScale
     #else
-        /**
-        A dictionary containing the URLs to image representations of this component.
+    /**
+     The scale factor of the image.
      
-        The keys of this dictionary are screen scale factors. The values of this dictionary are URLs to remote images that you should prefer over the `text` property once they are available. Use the URL that is best suited to the current screen’s scale factor.
-        */
-        public var imageURLs: [NSNumber: URL] = [:]
+     If you multiply the logical size of the image (stored in the `size` property) by the value in this property, you get the dimensions of the image in pixels.
+     
+     The default value of this property matches the natural scale factor associated with the main screen. However, only images with a scale factor of 1.0, 2.0, or 3.0 are ever returned by the API, so a scale factor of 1.0 of less results in a 1× (standard-resolution) image, while a scale factor greater than 1.0 results in a 2× (high-resolution or Retina) image.
+     */
+    open var scale: CGFloat = UIScreen.main.scale
```

Comment:
Oh, I didn’t mean to imply that the entire set of properties needed to be copied from MapboxStatic.swift, just that we would choose a scale automatically using `UIScreen.main.scale` or whatever the macOS or watchOS equivalent would be.

I think you can safely inline this code, but if you think it’s important to make the scale configurable, then a better way would be to replace `imageURL` with a method that takes the scale as an argument. Because this class doesn’t need to remember the scale.

Reply:
#213 is what I meant by inlining the code.